include/wil/registry_basic.h

```diff
@@ -962,28 +954,86 @@ namespace reg_view_details
                     continue;
                 }
 
-                return err_policy::HResult(HRESULT_FROM_WIN32(error));
+                return get_err_policy::HResult(HRESULT_FROM_WIN32(error));
             }
 
             ::wil::registry::reg_view_details::trim_buffer(return_value);
-            return err_policy::HResult(S_OK);
+            return get_err_policy::HResult(S_OK);
+        }
+
+    public:
+
+        template <typename R>
+        typename err_policy::result get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        {
+            return get_value_internal(subkey, value_name, return_value, type);
         }
 
         template <typename R>
-        typename err_policy::result get_value(_In_opt_ PCWSTR value_name, optional_value<R>& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        typename err_policy::result get_value(_In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
         {
             return get_value(nullptr, value_name, return_value, type);
         }
 
         // intended for err_exception_policy as err_returncode_policy will not get an error code
         template <typename R>
-        optional_value<R> get_value(_In_opt_ PCWSTR value_name) const
+        R get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
         {
-            optional_value<R> value;
-            get_value(value_name, value);
+            R value{};
+            get_value(subkey, value_name, value, type);
             return value;
         }
 
+        // intended for err_exception_policy as err_returncode_policy will not get an error code
+        template <typename R>
+        R get_value(_In_opt_ PCWSTR value_name, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        {
+            return get_value(nullptr, value_name, type);
+        }
+
+        template <typename R>
+        typename err_policy::result try_get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, optional_value<R>& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        {
+            const auto hr = get_value_internal<R, err_returncode_policy>(subkey, value_name, return_value.value, type);
+            if (SUCCEEDED(hr))
+            {
+                return_value.type = optional_value_status::has_value;
+                // TODO: set value

```

Comment:
I think the value has already been set at this point, so we may not have any other action to take.